src/safeposix/syscalls/net_calls.rs

```diff
@@ -159,35 +187,64 @@ impl Cage {
 
     //creates a sockhandle if none exists, otherwise this is a no-op
     pub fn force_innersocket(sockhandle: &mut SocketHandle) {
+        //The innersocket is an Option wrapped around the raw sys fd
+        //Depending on domain, innersocket may or may not be necessary
+        //Ex: Unix sockets do not have a kernal fd
+        //If innersocket is not available, then create a socket and insert its
+        //fd into innersocket
         if let None = sockhandle.innersocket {
+            //Create a new socket, as no sockhandle exists
+            //Socket creation rarely fails except for invalid parameters or extremely low-resources conditions
+            //Upon failure, process will panic
+            // ** What domains does lind satisfy???? ** //
             let thissock =
                 interface::Socket::new(sockhandle.domain, sockhandle.socktype, sockhandle.protocol);
 
+            //Loop through socket options and check which ones are set
+            //This is necessary as we can only set one option at a time
             for reuse in [SO_REUSEPORT, SO_REUSEADDR] {
+                //If socket option is not set, continue to next socket option
                 if sockhandle.socket_options & (1 << reuse) == 0 {
                     continue;
                 }
+
+                //Otherwise, set the socket option in the new socket
+                //The level argument specifies the protocol level at which the option resides
+                //In our case, we are setting options at the socket level
+                //To learn more about setsockopt https://man7.org/linux/man-pages/man3/setsockopt.3p.html
                 let sockret = thissock.setsockopt(SOL_SOCKET, reuse, 1);
+                //Failure occured upon setting a socket option
+                //Possible failures can be read at the man page linked above
+                //
+                // **Possibly add errors instead of having a single panick for all errors ??? ** //
                 if sockret < 0 {
                     panic!("Cannot handle failure in setsockopt on socket creation");
                 }
             }
 
+            //Insert the socket file descriptor into the innersocket value
             sockhandle.innersocket = Some(thissock);
         };
     }
 
     //we assume we've converted into a RustSockAddr in the dispatcher
+    //bind a name to a socket - assign the address specified by localaddr to the 
+    //socket referred to by the file descriptor fd
```

Comment:
The bind_inner function seems pointless.  Why have it?

Reply:
I'm pretty sure this is a remnant from a refactor at some point.